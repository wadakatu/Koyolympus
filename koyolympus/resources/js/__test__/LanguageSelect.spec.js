import {shallowMount} from '@vue/test-utils';
import Component from '@/LanguageSelectComponent.vue';

describe('Testing method', () => {
    test('Testing changeLang method', () => {
        const wrapper = shallowMount(Component);
        wrapper.vm.changeLang();
        expect(wrapper.emitted('update')).not.toBeUndefined();
        expect(wrapper.emitted('update')[0][0]).toBe('japanese');
    });
});

describe('Testing @ event', () => {
    let wrapper;
    let options;
    beforeEach(() => {
        wrapper = shallowMount(Component);
        wrapper.vm.changeLang = jest.fn();
        options = wrapper.find('select').findAll('option');
    });
    describe('Testing change event', () => {
        test('click Japanese', async () => {
            await options.at(1).setSelected();
            await options.at(0).setSelected();
            expect(wrapper.vm.language).toBe('japanese');
            expect(wrapper.vm.changeLang).toHaveBeenCalled();
        });
        test('click English', async () => {
            await options.at(1).setSelected();
            expect(wrapper.vm.language).toBe('english');
            expect(wrapper.vm.changeLang).toHaveBeenCalled();
        });
        test('click French', async () => {
            await options.at(2).setSelected();
            expect(wrapper.vm.language).toBe('french');
            expect(wrapper.vm.changeLang).toHaveBeenCalled();
        });
        test('click Korean', async () => {
            await options.at(3).setSelected();
            expect(wrapper.vm.language).toBe('korean');
            expect(wrapper.vm.changeLang).toHaveBeenCalled();
        });
        test('click Chinese', async () => {
            await options.at(4).setSelected();
            expect(wrapper.vm.language).toBe('chinese');
            expect(wrapper.vm.changeLang).toHaveBeenCalled();
        });
    });
});

describe('Testing snapshot', () => {
    test('default', () => {
        const wrapper = shallowMount(Component);
        expect(wrapper.element).toMatchSnapshot();
    });
});
