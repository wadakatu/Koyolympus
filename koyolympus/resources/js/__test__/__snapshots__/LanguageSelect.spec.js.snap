// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing snapshot default 1`] = `
<div
  class="select"
  tabindex="1"
>
  <input
    checked="checked"
    class="selectopt"
    id="opt1"
    name="test"
    type="radio"
  />
   
  <label
    class="option"
    for="opt1"
  >
    日本語
  </label>
   
  <input
    class="selectopt"
    id="opt2"
    name="test"
    type="radio"
  />
   
  <label
    class="option"
    for="opt2"
  >
    English
  </label>
   
  <input
    class="selectopt"
    id="opt3"
    name="test"
    type="radio"
  />
   
  <label
    class="option"
    for="opt3"
  >
    French
  </label>
   
  <input
    class="selectopt"
    id="opt4"
    name="test"
    type="radio"
  />
   
  <label
    class="option"
    for="opt4"
  >
    한국말
  </label>
   
  <input
    class="selectopt"
    id="opt5"
    name="test"
    type="radio"
  />
   
  <label
    class="option"
    for="opt5"
  >
    中文
  </label>
</div>
`;
